### DISCUSSION TITLE
Isolated Event for Items on Equip, Unequip, Pickup/drop, use/onhit

### DISCUSSION THREAD

**Discussion by PyroTech03** [AscensionGameDev/Intersect-Engine#2181] [Ideas]
I would like to suggest allowing individual isolated events within the item editor similar to how we have isolated events within the quest editor for task completion. 

I believe at least we should have an event for Equip and Unequip.  But also having one for on Pickup and drop would be great too.

I think this would be potentially better as currently the best we have to emulate this is by doing a common event with the "On equipment/inventory change" and either have multiple of these as we need it for effects or letting one call get longer and longer. This method makes calls on every item and every player.  By allowing isolated events, we can reduce it and have it done only on items that actually need it done.

And another area would be an event on use/hit. This would allow us to attach events to items without being explicitly tied to an event type. This could allow us to add a skill system to weapons or armor for example.

**Comment by PyroTech03** [comment]
Complementary idea #2202

**Comment by Arufonsu** [comment]
Resolved by #2208
